terraform-provider-constellation/internal/provider/cluster_resource.go

```diff
@@ -352,29 +361,43 @@ func (r *ClusterResource) Configure(_ context.Context, req resource.ConfigureReq
 	}
 
 	r.newApplier = func(ctx context.Context, validator atls.Validator) *constellation.Applier {
-		return constellation.NewApplier(&tfContextLogger{ctx: ctx}, &nopSpinner{}, newDialer)
+		return constellation.NewApplier(&tfContextLogger{ctx: ctx}, &nopSpinner{}, constellation.ApplyContextTerraform, newDialer)
 	}
 }
 
 // ModifyPlan is called when the resource is planned for creation, updates, or deletion. This allows to set pre-apply
 // warnings and errors.
 func (r *ClusterResource) ModifyPlan(ctx context.Context, req resource.ModifyPlanRequest, resp *resource.ModifyPlanResponse) {
```

Comment:
See the PR description: [e2e test](https://github.com/edgelesssys/constellation/actions/runs/7274425264/job/19820350348)

Reply:
once the above PR is merged, we should run the example and not this test when we change provider configuration, since it can catch more errors. The above test runs a different setup and doesn't cover unknown values. But it's okay in this case.